src/lightning/fabric/fabric.py

```diff
@@ -971,7 +972,9 @@ def _validate_setup_module(self, module: nn.Module) -> None:
 
     def _validate_setup_optimizers(self, optimizers: Sequence[Optimizer]) -> None:
         self._validate_launched()
-        if isinstance(self._strategy, (DeepSpeedStrategy, XLAStrategy)):
+        if isinstance(self._strategy, (DeepSpeedStrategy, XLAStrategy)) and not isinstance(
+            self._strategy, XLAFSDPStrategy
```

Comment:
This would not necessary if XLAFSDPStrategy didn't inherit from XLAStrategy (see my other comment) :)

Reply:
However, the inverse of this check should move to line 961 where we validate that the strategy must setup model and optimizers separately (since the XLAFSDP strategy raises NotImiplementedError for the joint setup).

Reply:
Will resolve this comment since I removed `XLAFSDPStrategy` inheritance from `XLAStrategy` and made the adjustments here